python/cudf/cudf/_lib/copying.pyx

```diff
@@ -184,26 +184,21 @@ def scatter(list sources, Column scatter_map, list target_columns,
                 f"index out of bounds for column of size {n_rows}"
             )
 
-    if isinstance(sources[0], Column):
-        tbl = pylibcudf.copying.scatter_table(
-            pylibcudf.Table([col.to_pylibcudf(mode="read") for col in sources]),
-            scatter_map.to_pylibcudf(mode="read"),
-            pylibcudf.Table([col.to_pylibcudf(mode="read") for col in target_columns]),
-        )
-    else:
-        tbl = pylibcudf.copying.scatter_scalars(
-            [(<DeviceScalar> as_device_scalar(slr)).c_value for slr in sources],
-            scatter_map.to_pylibcudf(mode="read"),
-            pylibcudf.Table([col.to_pylibcudf(mode="read") for col in target_columns]),
-        )
+    tbl = pylibcudf.copying.scatter(
+        pylibcudf.Table([col.to_pylibcudf(mode="read") for col in sources])
```

Comment:
I'm confused, was this question meant for the other thread?

Reply:
No, I'm trying to ask a related question here. I realize now it's because this is `def` and the other is `cpdef`.

Reply:
Ah right that's why I was confused, now I see. You're on the right track; it's less about this function being a def or a cpdef, and more about what it's calling. We're calling pylibcudf as a Python API here, so it accepts arbitrary objects as input and we're not cdefing them. In the other places we have to cdef the inputs because we are calling cdef functions (C++ cdef extern functions) that require strongly typed input.